spec/cucumber/core/test/filters/locations_filter_spec.rb

```diff
@@ -102,20 +102,90 @@ module Core
                   | 1 | 2 |
                   | 3 | 4 |
 
+              Rule: A rule with a background
+                Background: A background rule
+                  Given background
+
+                Scenario: with a rule and background
+                  Given rule a
+
+                Scenario: another with a rule and background
+                  Given rule b
+
+              Rule: A rule without a background
+                Scenario: with a rule and no background
+                  Given rule c
+
+                Scenario: another with a rule and no background
+                  Given rule d
             FEATURE
           end
 
           def test_case_named(name)
             test_cases.find { |c| c.name == name }
           end
 
+          it 'matches the feature line to all scenarios' do
+            location = Test::Location.new(file, 1)
+            filter = described_class.new([location])
+            compile [doc], receiver, [filter]
+            expect(receiver.test_case_locations).to eq test_cases.map(&:location)
+          end
+
+          it 'matches the background line to all scenarios' do
+            location = Test::Location.new(file, 2)
+            filter = described_class.new([location])
+            compile [doc], receiver, [filter]
+            expect(receiver.test_case_locations).to eq test_cases.map(&:location)
+          end
+
           it 'matches the location on a background step to all scenarios' do
             location = Test::Location.new(file, 3)
             filter = described_class.new([location])
             compile([doc], receiver, [filter])
             expect(receiver.test_case_locations).to eq test_cases.map(&:location)
           end
 
+          it 'matches the rule with background line to all of its scenarios' do
+            location = Test::Location.new(file, 29)
+            filter = described_class.new([location])
+            compile [doc], receiver, [filter]
+            expect(receiver.test_case_locations).to eq [
+              test_case_named('with a rule and background').location,
+              test_case_named('another with a rule and background').location
+            ]
+          end
+
+          it 'matches the rule background line to all of the rules scenarios' do
+            location = Test::Location.new(file, 30)
+            filter = described_class.new([location])
+            compile [doc], receiver, [filter]
+            expect(receiver.test_case_locations).to eq [
+              test_case_named('with a rule and background').location,
+              test_case_named('another with a rule and background').location
+            ]
+          end
+
+          it 'matches the location on a rule background step to all of the rules scenarios' do
```

Comment:
not sure what we mean by location? Do we just mean any GWT boomf?

Reply:
Looks like we're using "line" and "location" interchangeably. It's a line from a human perspective, but since this is testing the `Test::LocationsFilter` class in isolation, the parameter is a `Test::Location`, and the return is a `Test::Location`, I guess we should normalize this to "location." I'm make a pass and try to bring consistency to all the spec names, including your feedback above.

Reply:
@luke-hill Okay, pushed another commit tightening up the spec nomenclature.